causalpy/pymc_experiments.py

```diff
@@ -11,1882 +11,108 @@
-            w1 = 1 / ps[t == 1]
-            w0 = 1 / (1 - ps[t == 0])
-        elif weighting_scheme == "robust":
-            p_of_t = np.mean(t)
-            w1 = p_of_t / (ps[t == 1])
-            w0 = (1 - p_of_t) / (1 - ps[t == 0])
-        else:
-            w1 = (1 - ps[t == 1]) * t[t == 1]
-            w0 = ps[t == 0] * (1 - t[t == 0])
-        fig, axs = plt.subplots(1, 2, figsize=(20, 6))
-        raw_trt = [
-            self.weighted_percentile(
-                X[t == 1][covariate].values, np.ones(len(X[t == 1])), p
-            )
-            for p in np.linspace(0, 1, 1000)
-        ]
-        raw_ntrt = [
-            self.weighted_percentile(
-                X[t == 0][covariate].values, np.ones(len(X[t == 0])), p
-            )
-            for p in np.linspace(0, 1, 1000)
-        ]
-        w_trt = [
-            self.weighted_percentile(X[t == 1][covariate].values, w1, p)
-            for p in np.linspace(0, 1, 1000)
-        ]
-        w_ntrt = [
-            self.weighted_percentile(X[t == 0][covariate].values, w0, p)
-            for p in np.linspace(0, 1, 1000)
-        ]
-        axs[0].plot(
-            np.linspace(0, 1, 1000), raw_trt, color="skyblue", label="Raw Treated"
-        )
-        axs[0].plot(
-            np.linspace(0, 1, 1000), raw_ntrt, color="orange", label="Raw Control"
-        )
-        axs[0].set_title(f"ECDF \n Raw: {covariate}")
-        axs[1].set_title(
-            f"ECDF \n Weighted {weighting_scheme} adjustment for {covariate}"
-        )
-        axs[1].plot(
-            np.linspace(0, 1, 1000), w_trt, color="skyblue", label="Reweighted Treated"
-        )
-        axs[1].plot(
-            np.linspace(0, 1, 1000),
-            w_ntrt,
-            color="orange",
-            label="Reweighted Control",
-        )
-        axs[1].set_xlabel("Quantiles")
-        axs[0].set_xlabel("Quantiles")
-        axs[1].legend()
-        axs[0].legend()
-        return fig
+from .experiments.instrumental_variable import (
+    InstrumentalVariable as NewInstrumentalVariable,
+)
+from .experiments.inverse_propensity_weighting import (
+    InversePropensityWeighting as NewInversePropensityWeighting,
+)
+from .experiments.prepostfit import (
+    InterruptedTimeSeries as NewInterruptedTimeSeries,
+)
+from .experiments.prepostfit import (
+    SyntheticControl as NewSyntheticControl,
+)
+from .experiments.prepostnegd import PrePostNEGD as NewPrePostNEGD
+from .experiments.regression_discontinuity import (
+    RegressionDiscontinuity as NewRegressionDiscontinuity,
+)
+from .experiments.regression_kink import RegressionKink as NewRegressionKink
+
+# Ensure deprecation warnings are always shown in Jupyter Notebooks
+warnings.simplefilter("always", DeprecationWarning)
+
+
+def PrePostNEGD(*args, **kwargs):
+    warnings.warn(
+        """causalpy.pymc_experiments.PrePostNEGD is deprecated and will be removed in a future release. Please use causalpy.experiments.PrePostNEGD instead.""",
```

Comment:
I can clarify with this as an example.

The recommendation here is that the user should use `causalpy.experiments.PrePostNEGD`. However, that isn't a real import. Also, most of your examples use `cp.PrePostNEGD` as you make the class available in the `causalpy/__init__.py`

What doesn't work: 
```python
# Recommended and doesn't work
import causalpy

causalpy.experiments.PrePostNEGD

# Also doesn't work
import causalpy.experiments

causalpy.experiments.PrePostNEGD

# Works but not recommended and probably not desired solution
from causalpy.experiments.prepostnegd import PrePostNEGD

PrePostNEGD


# Works and is used in the documentation
import causalpy as cp

cp.PrePostNEGD

# Or also since same but without alias
import causalpy

causalpy.PrePostNEGD
```

So just recommend either of the last two for all of these deprecations. Or import these in the `causalpy/experiments/__init__.py` in order to make this recommendation work

Reply:
Thanks, now I get it :)

Fixed in https://github.com/pymc-labs/CausalPy/pull/381/commits/01ce582f61468df45f2cde0479391389b9c28985. I've also updated it so that the depreciation warning text shows up as red in the terminal output, so should be more visible.